### PR TITLE
[generic-worker] Add support for mounting tar.xz and tar.zst files.

### LINKED ISSUE #3540: Add support for .tar.zstd archives in generic-worker
This was begun in #3351 but ran into other issues.  It remains a useful feature that should be easy to add once those issues are resolved.

### PR DESCRIPTION
**Pull request by jschwartzentruber**
Revive #3351. This updates archiver to the latest version now that mholt/archiver/issues/213 is fixed. Add support for generic-worker mounts in `tar.xz` and `tar.zst` formats.

Github Bug/Issue: Fixes #3540

### CODE REVIEW

**Comment by petemoore** [comment]
Above test failures seem to emanate from https://github.com/mholt/archiver/blob/v3.5.1/targz.go#L102. Error is:

```
runtime error: invalid memory address or nil pointer dereference
```

This suggests either `tgz.Tar` is `nil` or `tgz` is `nil`.

**Comment by petemoore** [comment]
Stack trace:

```
panic({0xdac800, 0x141a6a0})
	/task_164501024657790/go1.17.7/go/src/runtime/panic.go:1047 +0x266
github.com/mholt/archiver/v3.(*TarGz).wrapReader(0xc00020a060)
	/task_164501024657790/gopath1.17.7/pkg/mod/github.com/mholt/archiver/v3@v3.5.1/targz.go:102 +0x11e
github.com/mholt/archiver/v3.(*TarGz).Unarchive(0xc00020a060, {0xc0003c4240, 0x59}, {0xc0001c2180, 0x7e})
	/task_164501024657790/gopath1.17.7/pkg/mod/github.com/mholt/archiver/v3@v3.5.1/targz.go:52 +0x45
github.com/taskcluster/taskcluster/v44/workers/generic-worker.extract({0xfd6e40, 0xc000430260}, {0xc00059890a, 0xe5b6e5}, {0xc0001c2180, 0x7e}, 0xc0002c6000)
	/task_164501024657790/taskcluster/workers/generic-worker/mounts.go:679 +0x557
```

**Comment by petemoore** [comment]
With logging we see:

```
Extracting tar.gz file /task_164501745431996/taskcluster/workers/generic-worker/downloads/TRMImvTERLavnufOUCy_5A to '/task_164501745431996/taskcluster/workers/generic-worker/testdata/TestMounts/tasks/task_164501745431996/my-task-caches/package'
```

which suggests that the archiver call that is made is:

```go
archiver.TarGz{}.Unarchive("/task_164501745431996/taskcluster/workers/generic-worker/downloads/TRMImvTERLavnufOUCy_5A", "/task_164501745431996/taskcluster/workers/generic-worker/testdata/TestMounts/tasks/task_164501745431996/my-task-caches/package")
```

With this construction, we find the reason for the `nil pointer dereference`: `archiver.TarGz{}.Tar` is `nil`.

**Comment by petemoore** [comment]
Note, we should also wait for #5190 to land before we merge this in.

File: go.mod
Comment (petemoore) [inline go.mod:59]: I'm not so happy that bumping versions of some dependencies results in the downgrading of other indirect dependencies. :-/

File: changelog/IcspTTolTAeaYCJWYDikYA.md
Comment (matt-boris) [inline changelog/IcspTTolTAeaYCJWYDikYA.md:3]: Could we delete this changelog?

**Review by matt-boris — APPROVED** [review]

File: changelog/IcspTTolTAeaYCJWYDikYA.md
Comment (petemoore) [reply changelog/IcspTTolTAeaYCJWYDikYA.md:3]: I think we'll get an error in the CI if we do that.

Comment (matt-boris) [reply changelog/IcspTTolTAeaYCJWYDikYA.md:3]: Even if we have this one? `changelog/issue-3540.md`

https://github.com/taskcluster/taskcluster/blob/c5e32265fc6d624a141de9d1fd1b4cbd8ab363ec/changelog/issue-3540.md

Comment (petemoore) [reply changelog/IcspTTolTAeaYCJWYDikYA.md]: Oh silly me, good point! 😆

**Comment by matt-boris** [comment]
`docker-worker-chunk-2` was successful on a rerun https://community-tc.services.mozilla.com/tasks/SiqgR6dTTeyPvaXOPic8qg

**Pull request merged**
